internal/service/cluster/model_cluster.go

```diff
@@ -71,27 +73,36 @@ func flattenPolicyItems(items []matlas.PolicyItem) []map[string]any {
 	return policyItems
 }
 
-func flattenProcessArgs(p *matlas.ProcessArgs) []map[string]any {
+func flattenProcessArgs(p20240530 *admin20240530.ClusterDescriptionProcessArgs, p *admin.ClusterDescriptionProcessArgs20240805) []map[string]any {
 	flattenedProcessArgs := []map[string]any{
 		{
-			"default_read_concern":                 p.DefaultReadConcern,
+			// default_read_concern and fail_index_key_too_long have been deprecated, hence using the older SDK
+			"default_read_concern":                 p20240530.DefaultReadConcern,
+			"fail_index_key_too_long":              p20240530.FailIndexKeyTooLong,
 			"default_write_concern":                p.DefaultWriteConcern,
-			"fail_index_key_too_long":              cast.ToBool(p.FailIndexKeyTooLong),
-			"javascript_enabled":                   cast.ToBool(p.JavascriptEnabled),
-			"minimum_enabled_tls_protocol":         p.MinimumEnabledTLSProtocol,
-			"no_table_scan":                        cast.ToBool(p.NoTableScan),
+			"javascript_enabled":                   p.JavascriptEnabled,
+			"minimum_enabled_tls_protocol":         p.MinimumEnabledTlsProtocol,
+			"no_table_scan":                        p.NoTableScan,
 			"oplog_size_mb":                        p.OplogSizeMB,
 			"oplog_min_retention_hours":            p.OplogMinRetentionHours,
 			"sample_size_bi_connector":             p.SampleSizeBIConnector,
 			"sample_refresh_interval_bi_connector": p.SampleRefreshIntervalBIConnector,
 			"transaction_lifetime_limit_seconds":   p.TransactionLifetimeLimitSeconds,
+			"tls_cipher_config_mode":               p.TlsCipherConfigMode,
+			"custom_openssl_cipher_config_tls12":   p.GetCustomOpensslCipherConfigTls12(),
 		},
 	}
-	if p.ChangeStreamOptionsPreAndPostImagesExpireAfterSeconds != nil {
-		flattenedProcessArgs[0]["change_stream_options_pre_and_post_images_expire_after_seconds"] = p.ChangeStreamOptionsPreAndPostImagesExpireAfterSeconds
-	} else {
+
+	if v := p.ChangeStreamOptionsPreAndPostImagesExpireAfterSeconds; v == nil {
 		flattenedProcessArgs[0]["change_stream_options_pre_and_post_images_expire_after_seconds"] = -1 // default in schema, otherwise user gets drift detection
```

Comment:
i don't know if we can avoid this, as we can have problems in TPF

Reply:
Here the SDK was updated from manual one to Atlas Go SDK but other than that I have refrained from making any updates other than to cipher config attributes in this PR.
Would not want to include other attributes at this point due to the urgent requirement & limited time after MMS release of this feature.